ramalama/http_client.py

```diff
@@ -60,7 +60,7 @@ def perform_download(self, file, progress):
         self.now_downloaded = 0
         self.start_time = time.time()
         while True:
-            data = self.response.read(1024)
+            data = self.response.read(100 * 1024)
```

Comment:
The patch looks good to me. How about we add a constant 100KB=100*1024 and use it here? It's faster reading for all people.

Reply:
I'm not sure this is the best fix, I think this works because the read is so big it delays the time that the next iteration of the loop is caused. In practice you cannot depend on TCP packet sizes being larger than 1400 bytes (because of middleboxes, etc.) and this encourages fragmenting the bytes, which isn't spectacular for reliability, performance, etc.

I think it's like the whole printing the progress bar thing is spiking the CPU usage, so if we just avoid updating that I think it could be a better fix to try (just updated this PR): 

https://github.com/containers/ramalama/pull/717

But I don't have CPU spike issues personally so cannot test the above PR.

Reply:
But reading python http library, they seem to default to 8kb, that seems more reasonable than 100kb:

https://github.com/python/cpython/blob/main/Lib/http/client.py

Reply:
We could try both PRs together

Reply:
I'm happy to drop #717 if it doesn't make much difference to CPU usage, although it does seem excessive to update the progress bar on each iteration.

Reply:
> I'm not sure this is the best fix, I think this works because the read is so big it delays the time that the next iteration of the loop is called. In practice you cannot depend on TCP packet sizes being larger than 1400 bytes (because of middleboxes, legacy, etc.) and this encourages fragmenting the bytes, which isn't spectacular for reliability, performance (from the transport perspective), etc.

As far as I understand the Python http client library code, read() does not operate at TCP level but at HTTP stream level, right? That would render this concern moot.

Reply:
True, but it's still a consideration as HTTP is on top of TCP

Reply:
IMHO, Splitting in two patches make sense, one for the progress bar and another for buffer size to be discussed. The progress bar skipping should already help as @ericcurtin mentioned.